Traceback (most recent call last):
  File "/Users/yewang/.local/pipx/venvs/jupyter-book/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/yewang/.local/pipx/venvs/jupyter-book/lib/python3.9/site-packages/nbclient/client.py", line 1314, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/yewang/.local/pipx/venvs/jupyter-book/lib/python3.9/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
  File "/Users/yewang/.pyenv/versions/3.9.15/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/Users/yewang/.local/pipx/venvs/jupyter-book/lib/python3.9/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/Users/yewang/.local/pipx/venvs/jupyter-book/lib/python3.9/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/yewang/.local/pipx/venvs/jupyter-book/lib/python3.9/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
from sklearn.ensemble import IsolationForest

# 生成示例数据
rng = np.random.RandomState(42)
X = 0.3 * rng.randn(100, 2)
X_train = np.r_[X + 2, X - 2]
X_outliers = rng.uniform(low=-4, high=4, size=(20, 2))
X = np.r_[X_train, X_outliers]

# 拟合孤立森林模型
clf = IsolationForest(random_state=0)
clf.fit(X)
y_pred = clf.predict(X)

# 绘制异常检测结果
plt.scatter(X[:, 0], X[:, 1], c=y_pred, cmap='viridis')
plt.title('孤立森林异常检测示例')
plt.xlabel('特征 1')
plt.ylabel('特征 2')
plt.show()
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mModuleNotFoundError[0m                       Traceback (most recent call last)
Cell [0;32mIn[1], line 1[0m
[0;32m----> 1[0m [38;5;28;01mfrom[39;00m [38;5;21;01msklearn[39;00m[38;5;21;01m.[39;00m[38;5;21;01mensemble[39;00m [38;5;28;01mimport[39;00m IsolationForest
[1;32m      3[0m [38;5;66;03m# 生成示例数据[39;00m
[1;32m      4[0m rng [38;5;241m=[39m np[38;5;241m.[39mrandom[38;5;241m.[39mRandomState([38;5;241m42[39m)

[0;31mModuleNotFoundError[0m: No module named 'sklearn'

